Code/GraphMol/RGroupDecomposition/RGroupDecompData.h

```diff
@@ -458,24 +466,34 @@ struct RGroupDecompData {
     }
 
     if (ties.size() > 1) {
-      size_t min_perm_value = 0;
+      size_t max_perm_value = 0;
       size_t smallest_added_rgroups = labels.size();
+      std::vector<int> largest_heavy_counts(labels.size(), 0);
+      std::vector<int> ordered_labels;
+      std::copy_if(labels.begin(), labels.end(),
+                   std::back_inserter(ordered_labels),
+                   [](const int &i) { return !(i < 0); });
+      std::copy_if(labels.begin(), labels.end(),
+                   std::back_inserter(ordered_labels),
+                   [](const int &i) { return (i < 0); });
       for (const auto &tied_permutation : ties) {
```

Comment:
I think one of the issues we are facing is that there is a lot of complex code internal to the functions that can't really be regression tested on the outside.  We may consider making a utility set of functions that we can test outside the full code base.  This might be one of them.

Not for this PR though.

Reply:
This would be one that I would test independently :)